prov/sockets/src/sock_fabric.c

```diff
@@ -634,6 +622,62 @@ int sock_node_getinfo(const char *node, const char *service, uint64_t flags,
 	return ret;
 }
 
+static int sock_match_src_addr(struct slist_entry *entry, const void *src_addr)
```

Comment:
Yeah, it's passed in <code>sock_addr_matches_interface()</code> as a function pointer of type <code>slist_func_t(struct slist_entry *item, const void *arg)</code>